In [4]:
import pandas as pd

path = '/home/eltito/repositorios/Curso-NumPy-Pandas-Matplotlib/Online_Retail.csv'
df = pd.read_csv(path, encoding='latin1')

# Esto es una manipulación, puedo crear una columna nueva a partir de otras
df['TotalPrice'] = df['Quantity'] * df['UnitPrice']
df.head()

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country,TotalPrice
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/10 8:26,2.55,17850.0,United Kingdom,15.30
1,536365,71053,WHITE METAL LANTERN,6,12/1/10 8:26,3.39,17850.0,United Kingdom,20.34
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/10 8:26,2.75,17850.0,United Kingdom,22.00
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/10 8:26,3.39,17850.0,United Kingdom,20.34
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/10 8:26,3.39,17850.0,United Kingdom,20.34


In [5]:
df['HighValue'] = df['TotalPrice'] > 100

# La columna HighValue trae True si TotalPrice es > 100
df['HighValue'].value_counts()

HighValue
False    528355
True      13554
Name: count, dtype: int64

In [6]:
# Con esto traigo información sobre el df
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 10 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  str    
 1   StockCode    541909 non-null  str    
 2   Description  540455 non-null  str    
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  str    
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  str    
 8   TotalPrice   541909 non-null  float64
 9   HighValue    541909 non-null  bool   
dtypes: bool(1), float64(3), int64(1), str(5)
memory usage: 37.7 MB


Vemos que:

4 InvoiceDate 541909 non-ull float64

esto es una fecha, vamos a darle formato de fecha

In [7]:
df['InvoiceDate'] = pd.to_datetime(df['InvoiceDate'])
df.info()

/tmp/ipykernel_1741/40610843.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['InvoiceDate'] = pd.to_datetime(df['InvoiceDate'])


<class 'pandas.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 10 columns):
 #   Column       Non-Null Count   Dtype         
---  ------       --------------   -----         
 0   InvoiceNo    541909 non-null  str           
 1   StockCode    541909 non-null  str           
 2   Description  540455 non-null  str           
 3   Quantity     541909 non-null  int64         
 4   InvoiceDate  541909 non-null  datetime64[us]
 5   UnitPrice    541909 non-null  float64       
 6   CustomerID   406829 non-null  float64       
 7   Country      541909 non-null  str           
 8   TotalPrice   541909 non-null  float64       
 9   HighValue    541909 non-null  bool          
dtypes: bool(1), datetime64[us](1), float64(3), int64(1), str(4)
memory usage: 37.7 MB


In [8]:
df['DiscountedPrice'] = df['UnitPrice'].apply(lambda x: x*0.9)
df.head(3)

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country,TotalPrice,HighValue,DiscountedPrice
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,2010-12-01 08:26:00,2.55,17850.0,United Kingdom,15.30,False,2.295
1,536365,71053,WHITE METAL LANTERN,6,2010-12-01 08:26:00,3.39,17850.0,United Kingdom,20.34,False,3.051
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,2010-12-01 08:26:00,2.75,17850.0,United Kingdom,22.00,False,2.475


In [ ]:
# Una funcion de categorización que se aplica a una columna
def categorize_price(price):
    if price > 50:
        return 'High'
    elif price >= 20:
        return 'Medium'
    else:
        return 'Low'

df['PriceCategory'] = df['UnitPrice'].apply(categorize_price)

# Puedo usar esta columna creada y contar cuantos son por cada valor
df['PriceCategory'].value_counts()

PriceCategory
Low       537734
Medium      2902
High        1273
Name: count, dtype: int64